### ISSUE TITLE
`prop in dict` does not report error if `prop` is union type such as `MyClass | string`

### ISSUE THREAD

**Issue by mmorearty** [microsoft/TypeScript#36055]
**TypeScript Version:**  3.8.0-dev.20200104

**Search Terms:**
is:issue is:open in union in:title 
[This was hard to search for since "in" is such a common word]

**Code**

```ts
class Skipped { }

declare const skipped: Skipped;
declare const skippedOrString: Skipped | string;

const dict = { "a": 1, "b": 2, "c": 3 };

console.log("a" in dict);
console.log(skipped in dict);
console.log(skippedOrString in dict);
```

**Expected behavior:**

Should get a compile error on the last line (as well as the line before that), since `skippedOrString` is of type `Skipped | string`, where `Skipped` is a class, and thus can't be a dictionary key.

**Actual behavior:**

The second-to-last line does report an error, as it should; but the last line does not.

**Playground Link:**

http://www.typescriptlang.org/play/index.html?ts=3.8.0-dev.20200104&ssl=1&ssc=1&pln=11&pc=1#code/MYGwhgzhAEDKDWBLADsgpgE2gb2gXwCgCM1QwAnNaYAewDsIAXaCJVTALjjfQwG5ipcJWr0mLHpgDy5WI3KI6Acy4IUvaAB8W8xUoEFaDZhkTBmAXhzQARGBtcAjABpbAIwfQATK5vBPAMz4BkYQNCBoAHQgNEoAFHY20IrQpuYAlAKh4VEx8azqmMl0qWaMmYZiOdGxcQXsGDJyCsrFpRkCQA

**Related Issues:**
